icefoganalytics/internal-data-portal
# System Admin: Part 1: Edit/Update User Profile

Relates to:
- [0001, Data Modeling, 2024-01-05](https://docs.google.com/document/d/1Mdbl_Cy0BgKiq0DGdvOKUqJn-FdgbF0LSVykwf__FY8/edit?pli=1)
- [0002, UI Modeling, 2024-01-08](https://docs.google.com/document/d/1tP63Ob4XPBGlASPHVf1GnNb3hlIkybmWBqJDA61u3J0/edit)
- https://github.com/icefoganalytics/internal-data-portal/issues/66

# Context

Give admins the ability to create/update/delete users.
Give users the ability to update their own information.
Give users the ability to trigger sync, and turn sync on/off.

Stretch goals
Give users the ability to have a "primary" role.
Give users the ability to downgrade their role by setting a lower tier role to "primary"?


# Implementation

Build user profile page, so that users can update their own information.
Simplify sync pipeline naming, to increase future flexibility. Only the last step needs to know that sync interacts with the Yukon Government directory.

# Screenshots

User Profile
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/80be4288-5335-4ffc-b20b-2f12557c7e7e)

User Profile Edit
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/f55eadd5-b4b1-4624-8992-e03151707706)

User Profile Edit, with save buttons, showing new kebab menu
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/777411f8-afa7-4f41-8c39-4a5c8d249fe0)


# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Click the new top nav /profile link on your user name, or select the the /profile link from the kebab menu.
5. Check that your info looks correct and is read-only.
6. Note that the top nav /profile link is highlighted, as is the kebab link.
7. Click the "Edit" button to go to the /profile/edit page.
8. Check that you can see an edit interface for your user.
9. Note that the top nav /profile link is still highlighted, as is the kebab link.
10. Check that the "Back" button takes you back to the read-only view.
11. Go back to the edit page, and check that the "Cancel" button takes you back to the read-only view.
12. Go back to the edit page, check that you can update your user information and "Save".
    > Maybe save should return you to the read-only view?
13. Check that the top right "Sync" button works.
14. Check that the breadcrumbs are correct for the Profile "read" and "edit" pages.
